core/views.py

```diff
@@ -0,0 +1,256 @@
+        self.set_test_cookie()
+        return super(SignInView, self).get(request, *args, **kwargs)
+
+    @method_decorator(csrf_protect)
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        form_class = self.get_form_class()
+        form = self.get_form(form_class)
+
+        if form.is_valid():
+            self.check_and_delete_test_cookie()
+            return self.form_valid(form)
+        else:
+            self.set_test_cookie()
+            return self.form_invalid(form)
+
+
+class LogOutView(RedirectView):
+    url = "/"
+
+    def get(self, request, *args, **kwargs):
+        auth_logout(request)
+        return super(LogOutView, self).get(request, *args, **kwargs)
+
+
+class ConfirmEmailView(generic_views.FormView, LoginRequiredMixin):
+
+    form_class = hacker_forms.ConfirmEmailForm
+    template_name = hacker_forms.ConfirmEmailForm.template_name
+    success_url = hacker_forms.ConfirmEmailForm.success_url
+
+    def form_valid(self, form):
+        form.full_clean()
+        return super(ConfirmEmailView, self).form_valid(form)
+
+    def code_invalid(self, form):
+        return render(self.request, self.template_name, {"invalid_code": True})
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class(initial=self.initial)
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+        invalid_code = False
+        if form.is_valid():
+            email = form.cleaned_data.get("email")
+            code = form.cleaned_data.get("confirm_code")
+            hacker = hacker_models.Hacker.objects.get(email=email)
+            if hacker.confirm_email(code):
+                # given confirm_code is correct
+                hacker_models.Hacker.objects.filter(email=email).update(
+                    confirm_code=None
+                )
+                hacker_models.Hacker.objects.filter(email=email).update(
+                    email_confirmed=True
+                )
+                return redirect(reverse_lazy("apply"))
+            else:
+                # given confirm_code is NOT correct
+                # ...
+                return self.code_invalid(form)
+
+        FormErrors = json.loads(form.errors.as_json())
+        return render(
+            request,
+            self.template_name,
+            {"form": form, "FormErrors": FormErrors, "invalid_code": invalid_code},
+        )
+
+
+class CreateApplicationView(generic_views.CreateView, LoginRequiredMixin):
+    model = hacker_models.Application
+    success_url = reverse_lazy("status")
+    template_used = "hacker/application_form.html"
+    fields = [
+        "major",
+        "gender",
+        "classification",
+        "grad_year",
+        "interests",
+        "essay",
+        "notes",
+    ]
+
+    grad_options = [op[0] for op in hacker_models.GRAD_YEARS]
```

Comment:
I'm guessing that `op` is supposed to mean `option`, let's go with `option` for clarity.